src/session.cpp

```diff
@@ -695,11 +701,16 @@ void Session::on_execute(uv_async_t* data) {
 
   bool is_closing = false;
 
-  RequestHandler* request_handler = NULL;
-  while (session->request_queue_->dequeue(request_handler)) {
-    if (request_handler != NULL) {
-      request_handler->set_query_plan(session->new_query_plan(request_handler->request(),
-                                                              request_handler->encoding_cache()));
+  RequestHandler* temp = NULL;
+  while (session->request_queue_->dequeue(temp)) {
+    RequestHandler::Ptr request_handler(temp);
```

Comment:
This could move down into the if and change the if condition to be `if (temp) {`.